crates/compiler/builtins/roc/List.roc

```diff
@@ -1466,3 +1467,15 @@ forEachTry! = \list, func! ->
 
                 Err err ->
                     Err err
+
+## Build a value from the contents of a list, using an effectful function.
+##
+## This is the same as [walk], except that the step function can have effects.
```

Comment:
Could we get a usage example?

Reply:
Is this an appropriate example?
```roc
total = List.walk! [1, 2, 3] 0 \sum, i ->
        nextSum = sum + i
        _ = Stdout.line! "Sum: $(nextSum |> Num.toStr)"
        nextSum
```
Most effectful functions have Results, so this just discards the Stdout result.

Reply:
It'd be better to have some infallible example, and use one that can error if we ever implement `walk_try!`. `Utc.now!` is `{} => I128`, we could use that?

```elm
fibonacciWithTimes = List.walk! [1, 2, 3] 0 \(fibonaccis, times), i ->
    start = Utc.now!
    fib = fibonacci i
    end = Utc.now!
    delta_nanos = Utc.delta_as_nanos start end

    (fibonaccis |> List.append fib, times |> List.append delta_nanos
    nextSum = sum + i
    _ = Stdout.line! "Sum: $(nextSum |> Num.toStr)"
    nextSum

```

If you'd prefer your example, that's fine to me. I'd just change it to "Sum up to $(i |> Num.toStr): $(nextSum |> Num.toStr)" because just "Sum: " is not accurate.

Reply:
I like `Utc.now!` much better. I wasn't wild about using a function which can error for the exact same reason.

Pushing up a total spherical cow example that uses `Utc.now!` - concise, and not using a function which can error.
```
now_multiples = List.walk! [1, 2, 3] [] \nums, i ->
        now = Utc.now! {} |> Utc.to_millis_since_epoch
        List.append nums (now * i)
```

Reply:
Looks good to me!